### DISCUSSION TITLE
regularization parameters cause huge bump in the loss function

### DISCUSSION THREAD

**Discussion by akosmaroy** [ourownstory/neural_prophet#1557] [Q&A - get help using NeuralProphet]
Hi, I have a question regarding the learning_rate parameter when creating a NeuralProphet instance. In my case, it seems that I get a 'bump' on my loss function at step 65 of the learning process (total epochs == 100), irrespective of the learning rate parameter. This seems somewhat counter intuitive. Can it be that there are learning rate parameters at some internal parts of the NerualProphet model?

My naive intuition would be that such a 'bump' would be due to using a learning rate that is too high and 'overshoots' the minimum. But if that was the case, using a lower learning rate should somewhat mitigate this, and would not have this 'bump' at the very same epoch / step in the process.

Please see my loss function at learning rate 1e-4, 1-e5 and 1-e6, all showing a visible 'bump' at step 65.

![image](https://github.com/ourownstory/neural_prophet/assets/333140/837d6d39-feba-41eb-831f-3038640d5604)
![image (1)](https://github.com/ourownstory/neural_prophet/assets/333140/da3db27f-b740-4640-9da2-30a556339d93)
![image (2)](https://github.com/ourownstory/neural_prophet/assets/333140/22df3230-e3f7-4d24-84d6-71ccaf1e35cb)

**Comment by akosmaroy** [comment]
after doing some more investigation, it seems that the bumps appears if one uses regularization parameters when calling model.add_lagged_regressor() or model.add_future_regressor(). setting 'trend_reg' for the model alaso seems to bring this effect to a lesser degree.
